### PR TITLE
Refactor benchmarks

### PR DESCRIPTION
**Pull request by daemyung**
## 🙏 Describe the pull request

- 벤치마크를 리펙토링 했습니다.
- 빠져있던 `CosineSimiliarty`를 추가했습니다.

## ✅ Checklist

- [x] Code follows the project's coding conventions and style.
- [x] Tests have been added or updated to cover the changes.
- [x] Documentation has been updated, if necessary.

### CODE REVIEW

File: benchmarks/benchmarker.py [hunk 1]
```diff
@@ -42,11 +43,11 @@ def print_scenarios():
     print(
         ", ".join(
             [
-                "adaptive-avg-pool2d",
+                "adaptive-avg-pool2d.png",
                 "argmax",
                 "batch-norm",
                 "conv2d",
-                "dropout",
+                "cosine-similarity" "dropout",
```
Comment (mejai1206) [inline benchmarks/benchmarker.py]: `,` 와 개행이 필요 할 것 같습니다.

File: benchmarks/benchmark_dropout.py [hunk 2]
```diff
@@ -20,13 +20,13 @@
...
-def bench_dropout_forward(p, inp_sz, ctx):
+def bench_dropout_forward(p, inp_sz, backend):
     inp = torch.randn(inp_sz, device="cuda")
 
-    if ctx == "torch":
+    if backend == "torch":
         return triton.testing.do_bench_cudagraph(lambda: torch.nn.functional.dropout(inp, p))
     else:
-        return triton.testing.do_bench_cudagraph(lambda: trident.function.dropout(inp, p))
+        return triton.testing.do_bench(lambda: trident.function.dropout(inp, p))
```
Comment (mejai1206) [inline benchmarks/benchmark_dropout.py:29]: 혹시 `droupout`만 `do_bench_cudagraph`가 아닌 `do_bench` 인 이유가 있을까요?

File: benchmarks/benchmarker.py [hunk 3]
```diff
@@ -68,14 +70,16 @@ def print_scenarios():
 
 
 def run_benchmarks(scenario, mode, show_plots):
-    if scenario == "adaptive-avg-pool2d":
+    if scenario == "adaptive-avg-pool2d.png":
```
Comment (kakao-steve-ai) [inline benchmarks/benchmarker.py]: 여기에만 png가 들어가는데, 오타인거 같습니다.

File: benchmarks/benchmark_dropout.py (hunk 2 above)
Comment (daemyung) [reply benchmarks/benchmark_dropout.py:29]: `cuda graph`가 `Dropout`만 캐치를 못합니다.

File: benchmarks/benchmarker.py [hunk 4]
```diff
@@ -42,10 +43,11 @@ def print_scenarios():
     print(
         ", ".join(
             [
-                "adaptive-avg-pool2d",
+                "adaptive-avg-pool2d.png",
```
Comment (kakao-steve-ai) [inline benchmarks/benchmarker.py]: 여기에만 png가 들어가는데, 오타인거 같습니다.

File: benchmarks/benchmarker.py (hunk 1 above)
Comment (daemyung) [reply benchmarks/benchmarker.py]: 수정했습니다.

File: benchmarks/benchmarker.py (hunk 3 above)
Comment (daemyung) [reply benchmarks/benchmarker.py]: 수정했습니다.

File: benchmarks/benchmarker.py (hunk 4 above)
Comment (daemyung) [reply benchmarks/benchmarker.py]: 수정했습니다.

**Review by kakao-steve-ai — APPROVED** [review]
LGTM

**Review by mejai1206 — APPROVED** [review]
LGTM
